packages/graphql/src/schema/generation/connection-where-input.ts

```diff
@@ -0,0 +1,223 @@
+    return connectionWhereInputType;
+}
+
+function shouldGenerateConnectionSortInput(
+    relationshipAdapter: RelationshipAdapter | RelationshipDeclarationAdapter
+): boolean {
+    if (relationshipAdapter.hasAnyProperties) {
+        return true;
+    }
+    if (!(relationshipAdapter.target instanceof UnionEntityAdapter)) {
+        return true;
+    }
+    return false;
+}
+
+export function withConnectionSortInputType({
+    relationshipAdapter,
+    composer,
+}: {
+    relationshipAdapter: RelationshipAdapter | RelationshipDeclarationAdapter;
+    composer: SchemaComposer;
+}): InputTypeComposer | undefined {
+    if (!shouldGenerateConnectionSortInput(relationshipAdapter)) {
+        return;
+    }
+    const typeName = relationshipAdapter.operations.connectionSortInputTypename;
+    if (composer.has(typeName)) {
+        return composer.getITC(typeName);
+    }
+    const fields = makeConnectionSortInputTypeFields({ relationshipAdapter });
+    if (!fields) {
+        return;
+    }
+
+    const connectionSortITC = composer.createInputTC({
+        name: typeName,
+        fields,
+    });
+    return connectionSortITC;
+}
+
+function makeConnectionSortInputTypeFields({
+    relationshipAdapter,
+}: {
+    relationshipAdapter: RelationshipAdapter | RelationshipDeclarationAdapter;
+}): InputTypeComposerFieldConfigMapDefinition | undefined {
+    const targetIsInterfaceWithSortableFields =
+        relationshipAdapter.target instanceof InterfaceEntityAdapter &&
+        relationshipAdapter.target.sortableFields.length;
+
+    const targetIsConcreteWithSortableFields =
+        relationshipAdapter.target instanceof ConcreteEntityAdapter && relationshipAdapter.target.sortableFields.length;
+
+    const fields: InputTypeComposerFieldConfigMapDefinition = {};
+    if (targetIsInterfaceWithSortableFields || targetIsConcreteWithSortableFields) {
+        fields["node"] = relationshipAdapter.target.operations.sortInputTypeName;
+    }
+
+    /*
+        We include all properties here to maintain existing behaviour.
+        In future sorting by arrays should become an aggregation sort because it sorts by the length of the array.
+    */
+    if (relationshipAdapter.hasAnyProperties) {
+        fields["edge"] = relationshipAdapter.operations.sortInputTypeName;
+    }
+
+    if (Object.keys(fields).length === 0) {
+        return undefined;
+    }
+    return fields;
+}
+
+function withRelationshipObjectType({
+    relationshipAdapter,
+    composer,
+}: {
+    relationshipAdapter: RelationshipAdapter | RelationshipDeclarationAdapter;
+    composer: SchemaComposer;
+}): ObjectTypeComposer {
+    const typeName = relationshipAdapter.operations.relationshipFieldTypename;
+    if (composer.has(typeName)) {
+        return composer.getOTC(typeName);
+    }
+    const relationshipObjectType = composer.createObjectTC({
+        name: typeName,
+        fields: { cursor: new GraphQLNonNull(GraphQLString), node: `${relationshipAdapter.target.name}!` },
```

Comment:
just wondering about the manual inclusion of `!` to `relationshipAdapter.target.name`, is that something we are OK with? Is there a way we can use the GraphQL library to make that nonnull?

Reply:
I've just moved this code, unfortunately we are not yet in condition to reference a GraphQL type.. but soon, if we get a bit of time for refactoring